### ISSUE TITLE
Please note the HIG comments about the status bar in your README

### ISSUE THREAD

**Issue by andrewroycarter** [kevingibbon/KGStatusBar#5]
Very well written library- but I'd hate to see this become a big trend among iOS apps. The status bar is there for a reason- and apps like twitter / mailbox are covering it up without good reason. A user should be able to see the time / battery percentage / etc at all times unless there's a very _good_ reason not to. We already have the navigationItem's prompt property for this purpose (though no body ever uses it).

Referring to the [HIG](http://developer.apple.com/library/ios/#documentation/userexperience/conceptual/mobilehig/UIElementGuidelines/UIElementGuidelines.html#//apple_ref/doc/uid/TP40006556-CH13-SW41)

Think twice before hiding the status bar if your app is not a game or full-screen media-viewing app. Although these apps might permanently hide the status bar, you should understand the ramifications of this design decision. Permanently hiding the status bar means that users must quit your app to find out, for example, whether they need to recharge their device.

Note that most iPad apps do not need to hide the status bar to gain extra space, because the status bar occupies such a small fraction of the screen. On iPad, the subtle appearance of the status bar does not compete with your app for the user’s attention. The small size of the status bar, combined with the slightly rounded corners of the app’s upper bar, make the status bar seem like part of the device background.

Consider hiding the status bar (and all other app UI) while people are actively viewing full-screen media. If you do this, be sure to allow people to retrieve the status bar (and appropriate app UI) with a single tap. Unless you have a very compelling reason to do so, it’s best to avoid defining a custom gesture to redisplay the status bar because users are unlikely to discover such a gesture or remember it.

Don’t create a custom status bar. Users depend on the consistency of the system-provided status bar. Although you might hide the status bar in your app, it’s not appropriate to create custom UI that takes its place.

...

**Comment by kevingibbon** [comment]
Thanks for the detailed analysis Andrew. I agree with many of your points. I put the library together because I see a lot of very useful circumstances to put a status message in the top bar. Definitely on the iPhone. It's not my place to try to dictate how people should use it. 

I suggest only displaying it for at most 2 seconds. But we will see how people end up using it. 

Cheers,
Kevin

**Comment by paulofierro** [comment]
We included the status bar in our app and used it to tell users we were logging in, loading data, etc. Never on screen for more than a few seconds and it worked really well in the context of our app.

After being in review for 5 weeks our app was finally rejected by Apple today due to violating the HIG. Specifically:

10.1: Apps must comply with all terms and conditions explained in the Apple iPhone Human Interface Guidelines and the Apple iPad Human Interface Guidelines

10.3: Apps that do not use system provided items, such as buttons and icons, correctly and as described in the Apple iPhone Human Interface Guidelines and the Apple iPad Human Interface Guidelines may be rejected

I just got off the phone with Apple and their reason for rejection was this component and this component alone. I referenced a few other apps that use similar techniques (e.g. Mailbox) and they stated these were also violations.

In short, using this component may result in your app being rejected. Such a shame.
